### ISSUE TITLE
Prevent trace logs from being shown if minimum logging level is set

### ISSUE THREAD

**Issue by erri120** [Nexus-Mods/NexusMods.App#250]
Trace logs are being printed to the console, regardless of [minimum logging level](https://github.com/Nexus-Mods/NexusMods.App/blob/e00bcba0b685d78c11b01c429ac417b41b03e7b4/src/NexusMods.App/Program.cs#L104). This results in a lot of console spam, which has a negative impact on performance.

**Comment by erri120** [comment]
After doing some debugging, these are my findings:

`Microsoft.Extensions.Logging` will as the `ILogger` whether the log level is enabled by comparing it with the minimum level. However, this minimum level is logger dependent. A log provider, like NLog, can have a `LoggerFilterRule` which specifies a different log level. Microsoft's rule selector will only use the global minimal level, as a last resort:

https://github.com/dotnet/runtime/blob/d0a287a56fda74f610e2ada174e64512dc596130/src/libraries/Microsoft.Extensions.Logging/src/LoggerRuleSelector.cs#L15-L21

![2023-04-13_12-18-44](https://user-images.githubusercontent.com/16577545/231729702-c095fc20-d00e-4013-b227-db45251e85d1.png)

This filter gets added here: https://github.com/NLog/NLog.Extensions.Logging/blob/fed2ba1fb6bf3713d7ee6349264cec7fe21a41fd/src/NLog.Extensions.Logging/Internal/RegisterNLogLoggingProvider.cs#L32-L36

The solution is to change the default `NLogProviderOptions`:

```csharp
var options = new NLogProviderOptions
{
    RemoveLoggerFactoryFilter = false
};

loggingBuilder.AddNLog(config, options);
```

**Comment by erri120** [comment]
Fixed in #280
